### DISCUSSION TITLE
dataframe is highly fragmented: tried to run a system dynamics file made by Vensim software

### DISCUSSION THREAD

**Discussion by sanket-mishra-profile** [quaquel/EMAworkbench#326] [Q&A] [answered]
![image](https://github.com/quaquel/EMAworkbench/assets/86866430/6a931597-97ec-42a0-99d0-ec63e87cf5ce)
could you let me know why this error has been shown?

**Comment by sanket-mishra-profile** [comment]
![image](https://github.com/quaquel/EMAworkbench/assets/86866430/ed4a05d5-3d02-4415-bbd7-485112d5b932)
I am facing multiple issues, one of which has been shown above
the other issue is shown in this screenshot, where when I changed a few parameters in the model, it showed list assignment index out of range, but these values were working fine in vensim model

**Comment by quaquel** [comment]
Please provide full details on the code giving rise to the error.

**Comment by EwoutH** [comment]
Looks like an error thrown by [pysd](https://pysd.readthedocs.io/) from the screenshot.

A [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) would be very useful to help you further with this issue.

Since this is likely not an issue with the workbench, I'm going to convert this to a Discussion.

**Comment by sanket-mishra-profile** [comment]
#I tried to run .mdl file using pysd
```jupyter-notebook
!pip -q install pySD
import numpy as np
import pandas as pd
import pysd
import matplotlib.pyplot as plt
``` 
```jupyter-notebook
vmodel = pysd.read_vensim('Query_file.mdl')
SD_result = vmodel.run()
```
#the values of LT1, LT2 parameters in the original model are 30 and 3 respectively
#in the next step I tried to change the parameter (LT1, LT2) values to 60 and 6 respectively 
```jupyter-notebook
SD_result = vmodel.run(params={ 'LT1':60, 'LT2':6})
``` 
#I again tried to change the parameter values to 30 and 3, but it showed me an error: **list assignment index out of range**
```jupyter-notebook
SD_result = vmodel.run(params={ 'LT1':30, 'LT2':3})
``` 
[Query_file.zip](https://github.com/quaquel/EMAworkbench/files/13623164/Query_file.zip)
#please let me know why is it happening with my code
#I have attached vensim model in the zip file for reference

**Reply by sanket-mishra-profile** [reply]
@EwoutH @quaquel .. please have a look

**Reply by quaquel** [reply, marked as answer]
this seems a pysd issue because there is no workbench code involved at all.

**Reply by sanket-mishra-profile** [reply]
@quaquel Alright. Thanks
